### DISCUSSION TITLE
Default FZF setup defaulting to blue

### DISCUSSION THREAD

**Discussion by evanpurkhiser** [ibhagwan/fzf-lua#1510] [Q&A] [answered]
I'm stuck trying to figure out why the dafult lua-fzf setup is blue.

I am using Lazy and have disabled all other plugins, reduced my config to absolutely nothing except Lazy + fzf-lua and it's still blue like this.

Running `fzf` in the terminal works fine and has no strange color scheme like this.

Posting this in case anyone else has seen it or has any ideas.

When I figure it out i'll also be sure to post the answer here incase anyone bumps up against this

<img width="881" alt="image" src="https://github.com/user-attachments/assets/6f9d2a61-28ea-4aee-a575-61a2e23579eb">

**Comment by ibhagwan** [comment]
That’s an issue in your config / Lazy, try with fzf-lua’s mini.sh (top of the README), the default colors he me is this:
![IMG_2637](https://github.com/user-attachments/assets/71b35283-2931-4fdb-a463-2d18e64be89d)

The issue could also stem from `FZF_DEFAULT_OPTS` so also check your shell variables.

Run the below to try a true default (Lazy has its own config different from Fzf-lua defaults):
```sh
sh -c "$(curl -s https://raw.githubusercontent.com/ibhagwan/fzf-lua/main/scripts/mini.sh)"
```

Btw, AFAIK Lazy creates its own fzf-lua color scheme based on your neovim colorscheme so try different coloschemes too, you can also try to disable this behavior with:
```lua
:FzfLua files fzf_colors=false
```
If the above solves your issue you can add `fzf_colors=false` to your setup.

**Reply by evanpurkhiser** [reply]
Seems like it might actually have to do with my terminal emulator. I've disabled my `.profile` and `.bashrc` and have used the mini.sh script (really cool!) and I'm still seeing this blue

Alacritty looks like this

<img width="690" alt="image" src="https://github.com/user-attachments/assets/716275a1-10b1-45cf-b2a6-2368c87d07f9">

Terminal.app looks like this

<img width="738" alt="image" src="https://github.com/user-attachments/assets/fe5d2e70-e4c4-4114-abfe-e9e8e95c2f2b">

**Reply by ibhagwan** [reply]
> Seems like it might actually have to do with my terminal emulator. I've disabled my .profile and .bashrc and have used the mini.sh script (really cool!) and I'm still seeing this blue

You’re inside neovim in the screenshot, your colorscheme is also changing the terminal colors, run `:FzfLua colorschemes` and see if this changes with the selected color scheme.

**Reply by evanpurkhiser** [reply]
<img width="418" alt="image" src="https://github.com/user-attachments/assets/70780589-9f5a-4a2a-99d3-5f9909799a03">

still blue

> You’re inside neovim in the screenshot

Using mini.sh runs sandboxed of my config though right. It is the default nvim colorscheme.

In Terminal.App seems ok.

<img width="682" alt="image" src="https://github.com/user-attachments/assets/f4695375-1a84-4622-a9db-db4ba5d49df8">

Let me dig into alacritty

**Reply by ibhagwan** [reply]
> Using mini.sh runs sandboxed of my config though right. It is the default nvim colorscheme.

Correct.

> Let me dig into alacritty

If you wanna run a quick test you can try my alacritty config (which sets the terminal colors too)
https://github.com/ibhagwan/dots/blob/master/.config/alacritty/alacritty.toml

**Reply by evanpurkhiser** [reply]
Your config does fix it. 

<img width="616" alt="image" src="https://github.com/user-attachments/assets/5298e10c-fef7-4f0a-8f51-2f8f2d57347e">

I'll find what the offending config is in mine https://github.com/evanpurkhiser/dots-personal/blob/main/common/platform-osx/alacritty/alacritty.toml

**Reply by evanpurkhiser** [reply, marked as answer]
It's something to do with the `alacritty-direct` termcap file.

**Reply by ibhagwan** [reply]
> It's something to do with the `alacritty-direct` termcap file.

https://unix.stackexchange.com/questions/597445/why-would-i-set-term-to-xterm-256color-when-using-alacritty

**Reply by evanpurkhiser** [reply]
yeah, the termcap file is installed and I do need to use it, since I make use of the direct color indexing.

```
$ infocmp alacritty-direct
#       Reconstructed via infocmp from file: /Users/evan/.local/share/terminfo/61/alacritty-direct
alacritty-direct|alacritty with direct color indexing,
        am, bce, hs, km, mc5i, mir, msgr, npc, xenl,
        colors#32767, cols#80, it#8, lines#24, pairs#32767,
        acsc=``aaffggiijjkkllmmnnooppqqrrssttuuvvwwxxyyzz{{||}}~~,
        bel=^G, bold=\E[1m, cbt=\E[Z, civis=\E[?25l,
        clear=\E[H\E[2J, cnorm=\E[?12l\E[?25h, cr=^M,
        csr=\E[%i%p1%d;%p2%dr, cub=\E[%p1%dD, cub1=^H,
        cud=\E[%p1%dB, cud1=^J, cuf=\E[%p1%dC, cuf1=\E[C,
        cup=\E[%i%p1%d;%p2%dH, cuu=\E[%p1%dA, cuu1=\E[A,
        cvvis=\E[?12;25h, dch=\E[%p1%dP, dch1=\E[P, dim=\E[2m,
        dl=\E[%p1%dM, dl1=\E[M, dsl=\E]2;\007, ech=\E[%p1%dX,
        ed=\E[J, el=\E[K, el1=\E[1K, flash=\E[?5h$<100/>\E[?5l,
        fsl=^G, home=\E[H, hpa=\E[%i%p1%dG, ht=^I, hts=\EH,
        ich=\E[%p1%d@, il=\E[%p1%dL, il1=\E[L, ind=^J,
        indn=\E[%p1%dS, initc@, invis=\E[8m,
        is2=\E[!p\E[?3;4l\E[4l\E>, kDC=\E[3;2~, kEND=\E[1;2F,
        kHOM=\E[1;2H, kIC=\E[2;2~, kLFT=\E[1;2D, kNXT=\E[6;2~,
        kPRV=\E[5;2~, kRIT=\E[1;2C, kb2=\EOE, kbs=\177, kcbt=\E[Z,
        kcub1=\EOD, kcud1=\EOB, kcuf1=\EOC, kcuu1=\EOA,
        kdch1=\E[3~, kend=\EOF, kent=\EOM, kf1=\EOP, kf10=\E[21~,
        kf11=\E[23~, kf12=\E[24~, kf13=\E[1;2P, kf14=\E[1;2Q,
        kf15=\E[1;2R, kf16=\E[1;2S, kf17=\E[15;2~, kf18=\E[17;2~,
        kf19=\E[18;2~, kf2=\EOQ, kf20=\E[19;2~, kf21=\E[20;2~,
        kf22=\E[21;2~, kf23=\E[23;2~, kf24=\E[24;2~,
        kf25=\E[1;5P, kf26=\E[1;5Q, kf27=\E[1;5R, kf28=\E[1;5S,
        kf29=\E[15;5~, kf3=\EOR, kf30=\E[17;5~, kf31=\E[18;5~,
        kf32=\E[19;5~, kf33=\E[20;5~, kf34=\E[21;5~,
        kf35=\E[23;5~, kf36=\E[24;5~, kf37=\E[1;6P, kf38=\E[1;6Q,
        kf39=\E[1;6R, kf4=\EOS, kf40=\E[1;6S, kf41=\E[15;6~,
        kf42=\E[17;6~, kf43=\E[18;6~, kf44=\E[19;6~,
        kf45=\E[20;6~, kf46=\E[21;6~, kf47=\E[23;6~,
        kf48=\E[24;6~, kf49=\E[1;3P, kf5=\E[15~, kf50=\E[1;3Q,
        kf51=\E[1;3R, kf52=\E[1;3S, kf53=\E[15;3~, kf54=\E[17;3~,
        kf55=\E[18;3~, kf56=\E[19;3~, kf57=\E[20;3~,
        kf58=\E[21;3~, kf59=\E[23;3~, kf6=\E[17~, kf60=\E[24;3~,
        kf61=\E[1;4P, kf62=\E[1;4Q, kf63=\E[1;4R, kf7=\E[18~,
        kf8=\E[19~, kf9=\E[20~, khome=\EOH, kich1=\E[2~,
        kind=\E[1;2B, kmous=\E[M, knp=\E[6~, kpp=\E[5~,
        kri=\E[1;2A, mc0=\E[i, mc4=\E[4i, mc5=\E[5i, meml=\El,
        memu=\Em, op=\E[39;49m, rc=\E8, rep=%p1%c\E[%p2%{1}%-%db,
        rev=\E[7m, ri=\EM, rin=\E[%p1%dT, ritm=\E[23m, rmacs=\E(B,
        rmam=\E[?7l, rmcup=\E[?1049l\E[23;0;0t, rmir=\E[4l,
        rmkx=\E[?1l\E>, rmm=\E[?1034l, rmso=\E[27m, rmul=\E[24m,
        rs1=\Ec, rs2=\E[!p\E[?3;4l\E[4l\E>, sc=\E7,
        setab=\E[%?%p1%{8}%<%t4%p1%d%e48\:2\:\:%p1%{65536}%/%d\:%p1%{256}%/%{255}%&%d\:%p1%{255}%&%d%;m,
        setaf=\E[%?%p1%{8}%<%t3%p1%d%e38\:2\:\:%p1%{65536}%/%d\:%p1%{256}%/%{255}%&%d\:%p1%{255}%&%d%;m,
        setb@, setf@,
        sgr=%?%p9%t\E(0%e\E(B%;\E[0%?%p6%t;1%;%?%p5%t;2%;%?%p2%t;4%;%?%p1%p3%|%t;7%;%?%p4%t;5%;%?%p7%t;8%;m,
        sgr0=\E(B\E[m, sitm=\E[3m, smacs=\E(0, smam=\E[?7h,
        smcup=\E[?1049h\E[22;0;0t, smir=\E[4h, smkx=\E[?1h\E=,
        smm=\E[?1034h, smso=\E[7m, smul=\E[4m, tbc=\E[3g,
        tsl=\E]2;, u6=\E[%i%d;%dR, u7=\E[6n,
        u8=\E[?%[;0123456789]c, u9=\E[c, vpa=\E[%i%p1%dd,
```

I'm still trying to pinpoint what exactly regressed.

**Reply by evanpurkhiser** [reply]
The termcap file changed between alacritty 0.13.2 and 0.14 which was recently released.

The old termcap file does not produce the blue colors.

Here's the diff

```diff
-#      Reconstructed via infocmp from file: /Users/evan/.local/share/terminfo/61/alacritty-direct-0.13.2
+#      Reconstructed via infocmp from file: /Users/evan/.local/share/terminfo/61/alacritty-direct
 alacritty-direct|alacritty with direct color indexing,
        am, bce, hs, km, mc5i, mir, msgr, npc, xenl,
-       colors#0, cols#80, it#8, lines#24, pairs#32767,
+       colors#32767, cols#80, it#8, lines#24, pairs#32767,
        acsc=``aaffggiijjkkllmmnnooppqqrrssttuuvvwwxxyyzz{{||}}~~,
        bel=^G, bold=\E[1m, cbt=\E[Z, civis=\E[?25l,
        clear=\E[H\E[2J, cnorm=\E[?12l\E[?25h, cr=^M,
        csr=\E[%i%p1%d;%p2%dr, cub=\E[%p1%dD, cub1=^H,
        cud=\E[%p1%dB, cud1=^J, cuf=\E[%p1%dC, cuf1=\E[C,
        cup=\E[%i%p1%d;%p2%dH, cuu=\E[%p1%dA, cuu1=\E[A,
        cvvis=\E[?12;25h, dch=\E[%p1%dP, dch1=\E[P, dim=\E[2m,
        dl=\E[%p1%dM, dl1=\E[M, dsl=\E]2;\007, ech=\E[%p1%dX,
        ed=\E[J, el=\E[K, el1=\E[1K, flash=\E[?5h$<100/>\E[?5l,
        fsl=^G, home=\E[H, hpa=\E[%i%p1%dG, ht=^I, hts=\EH,
        ich=\E[%p1%d@, il=\E[%p1%dL, il1=\E[L, ind=^J,
        indn=\E[%p1%dS, initc@, invis=\E[8m,
        is2=\E[!p\E[?3;4l\E[4l\E>, kDC=\E[3;2~, kEND=\E[1;2F,
        kHOM=\E[1;2H, kIC=\E[2;2~, kLFT=\E[1;2D, kNXT=\E[6;2~,
        kPRV=\E[5;2~, kRIT=\E[1;2C, kb2=\EOE, kbs=\177, kcbt=\E[Z,
        kcub1=\EOD, kcud1=\EOB, kcuf1=\EOC, kcuu1=\EOA,
        kdch1=\E[3~, kend=\EOF, kent=\EOM, kf1=\EOP, kf10=\E[21~,
        kf11=\E[23~, kf12=\E[24~, kf13=\E[1;2P, kf14=\E[1;2Q,
        kf15=\E[1;2R, kf16=\E[1;2S, kf17=\E[15;2~, kf18=\E[17;2~,
        kf19=\E[18;2~, kf2=\EOQ, kf20=\E[19;2~, kf21=\E[20;2~,
        kf22=\E[21;2~, kf23=\E[23;2~, kf24=\E[24;2~,
        kf25=\E[1;5P, kf26=\E[1;5Q, kf27=\E[1;5R, kf28=\E[1;5S,
        kf29=\E[15;5~, kf3=\EOR, kf30=\E[17;5~, kf31=\E[18;5~,
        kf32=\E[19;5~, kf33=\E[20;5~, kf34=\E[21;5~,
        kf35=\E[23;5~, kf36=\E[24;5~, kf37=\E[1;6P, kf38=\E[1;6Q,
        kf39=\E[1;6R, kf4=\EOS, kf40=\E[1;6S, kf41=\E[15;6~,
        kf42=\E[17;6~, kf43=\E[18;6~, kf44=\E[19;6~,
        kf45=\E[20;6~, kf46=\E[21;6~, kf47=\E[23;6~,
        kf48=\E[24;6~, kf49=\E[1;3P, kf5=\E[15~, kf50=\E[1;3Q,
        kf51=\E[1;3R, kf52=\E[1;3S, kf53=\E[15;3~, kf54=\E[17;3~,
        kf55=\E[18;3~, kf56=\E[19;3~, kf57=\E[20;3~,
        kf58=\E[21;3~, kf59=\E[23;3~, kf6=\E[17~, kf60=\E[24;3~,
        kf61=\E[1;4P, kf62=\E[1;4Q, kf63=\E[1;4R, kf7=\E[18~,
        kf8=\E[19~, kf9=\E[20~, khome=\EOH, kich1=\E[2~,
        kind=\E[1;2B, kmous=\E[M, knp=\E[6~, kpp=\E[5~,
        kri=\E[1;2A, mc0=\E[i, mc4=\E[4i, mc5=\E[5i, meml=\El,
        memu=\Em, op=\E[39;49m, rc=\E8, rep=%p1%c\E[%p2%{1}%-%db,
        rev=\E[7m, ri=\EM, rin=\E[%p1%dT, ritm=\E[23m, rmacs=\E(B,
        rmam=\E[?7l, rmcup=\E[?1049l\E[23;0;0t, rmir=\E[4l,
        rmkx=\E[?1l\E>, rmm=\E[?1034l, rmso=\E[27m, rmul=\E[24m,
        rs1=\Ec, rs2=\E[!p\E[?3;4l\E[4l\E>, sc=\E7,
        setab=\E[%?%p1%{8}%<%t4%p1%d%e48\:2\:\:%p1%{65536}%/%d\:%p1%{256}%/%{255}%&%d\:%p1%{255}%&%d%;m,
        setaf=\E[%?%p1%{8}%<%t3%p1%d%e38\:2\:\:%p1%{65536}%/%d\:%p1%{256}%/%{255}%&%d\:%p1%{255}%&%d%;m,
        setb@, setf@,
        sgr=%?%p9%t\E(0%e\E(B%;\E[0%?%p6%t;1%;%?%p5%t;2%;%?%p2%t;4%;%?%p1%p3%|%t;7%;%?%p4%t;5%;%?%p7%t;8%;m,
        sgr0=\E(B\E[m, sitm=\E[3m, smacs=\E(0, smam=\E[?7h,
        smcup=\E[?1049h\E[22;0;0t, smir=\E[4h, smkx=\E[?1h\E=,
        smm=\E[?1034h, smso=\E[7m, smul=\E[4m, tbc=\E[3g,
        tsl=\E]2;, u6=\E[%i%d;%dR, u7=\E[6n,
        u8=\E[?%[;0123456789]c, u9=\E[c, vpa=\E[%i%p1%dd,
```

I'm not completely sure I understand what changed here yet.

**Reply by evanpurkhiser** [reply]
I've gone ahead and opened an issue on Alacritty: https://github.com/alacritty/alacritty/issues/8289

I don't know enough about termcap files to understand what changed here.

**Reply by ibhagwan** [reply]
> I've gone ahead and opened an issue on Alacritty: [alacritty/alacritty#8289](https://github.com/alacritty/alacritty/issues/8289)
> 
> I don't know enough about termcap files to understand what changed here.

Serious investigation for a blue color, improbably would’ve just changed TERM and called it a day lol
